samtranslator/model/sam_resources.py

```diff
@@ -845,18 +846,77 @@ def to_cloudformation(self, **kwargs):
         :rtype: list
         """
         resources = []
+        managed_policy_map = kwargs.get("managed_policy_map", {})
         synthetics_canary = self._construct_synthetics_canary()
         resources.append(synthetics_canary)
 
         # A S3 Bucket resource will be added to the transformed template if the user doesn't provide an artifact
         # bucket to store canary results
-        if synthetics_canary.ArtifactS3Location is None:
+
+        artifact_bucket_name = ""
+        if self.ArtifactS3Location is None:
             s3bucket = self._construct_artifact_bucket()
             resources.append(s3bucket)
             synthetics_canary.ArtifactS3Location = {"Fn::Join": ["", ["s3://", {"Ref": s3bucket.logical_id}]]}
+            artifact_bucket_name = {"Ref": s3bucket.logical_id}
+
+        if self.Role is None:
+            role = self._construct_role(artifact_bucket_name, synthetics_canary.Name, managed_policy_map)
+            resources.append(role)
+            synthetics_canary.ExecutionRoleArn = role.get_runtime_attr("arn")
 
         return resources
 
+    def _construct_role(self, artifact_bucket_name, canary_name, managed_policy_map):
+        """Constructs a Role based on this SAM Canaries's Policies and ArtifactS3Location property.
+
+        :returns: the generated IAM Role
+        :rtype: model.iam.IAMRole
+        """
+
+        assume_role_policy_document = IAMRolePolicies.lambda_assume_role_policy()
+        role_attributes = self.get_passthrough_resource_attributes()
+
+        # add AWS managed policies if user has enabled VpcConfig or Tracing
+        managed_policy_arns = []
+        if self.VpcConfig:
+            managed_policy_arns.append(
+                ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaVPCAccessExecutionRole")
+            )
+        if self.Tracing:
+            managed_policy_name = get_xray_managed_policy_name()
+            managed_policy_arns.append(ArnGenerator.generate_aws_managed_policy_arn(managed_policy_name))
+
+        # if user has defined Policies property, those policies will be appended to this role
+        function_policies = ResourcePolicies(
+            {"Policies": self.Policies},
+            # No support for policy templates in the "core"
```

Comment:
the `No support for policy templates in the "core"`? Not too sure, it was included in the other places where `ResourcePolicies` is used so I included it here just in case